core/arch/arm/emit_utils.c

```diff
@@ -948,6 +948,11 @@ emit_indirect_branch_lookup(dcontext_t *dc, generated_code_t *code, byte *pc,
     APP(&ilist,
         INSTR_CREATE_ldr(dc, OPREG(DR_REG_R1),
                          OPND_TLS_FIELD(TLS_MASK_SLOT(ibl_code->branch_type))));
+    /* We need the mask load to have Aqcuire semantics to pair with the Release in
+     * update_lookuptable_tls() and avoid the reader here seeing a new mask with
+     * an old table.
+     */
+    APP(&ilist, INSTR_CREATE_dmb(dc, OPND_CREATE_INT(DR_DMB_ISH)));
```

Comment:
This only needs to order loads, so it can be a DMB ISHLD.

Reply:
OK, what about the instances of DMB ISH in arch_exports.h: e..g, https://github.com/DynamoRIO/dynamorio/blob/master/core/arch/arch_exports.h#L845?

I based these DMB ISH on what I observed g++ 9.2.1 generating for std::atomic<int>.load(std::memory_order_acquire) on ARM and it uses "LDR; DMB ISH".  It seems to use DMB ISH for everything: fetch_add(); store(); load().

Reply:
I put a note here on potentially relaxing those other barriers: https://github.com/DynamoRIO/dynamorio/issues/2502#issuecomment-617286457